### PR TITLE
Include parser for Polyspace static analysis tool

### PR DESCRIPTION
**Pull request by evahabeeballah**
This pull request is to include a parser for .csv result files generated from Polyspace tool.
This request is linked analysis-model pull request: [https://github.com/jenkinsci/analysis-model/pull/863](url)
Link for polyspace static analysis tool: https://de.mathworks.com/products/polyspace.html

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by uhafner — CHANGES_REQUESTED** [review]
Requires a release (or incremental version) of https://github.com/jenkinsci/analysis-model-api-plugin before we can merge this change. You can reference that version in the pom.xml if it has been published.

Regarding your questions:
> Is it possible to change the names of the severity categories to something else other than Low, Normal, High? For example: Error, Not-Run, Warning, Incomplete.

No this is not possible, all parsers share the same model and the same severities.

> Regarding the License for the addition of icons in Warnings-NG, what is expected to be provided if the icon was created based on the logo of the tool (Cropped and resized) ?

You need to use the logo license as starting point. What is the current license? If this license allows modifications that we simply can stick to it. Note: it would be helpful to use an SVG if possible.

**Comment by evahabeeballah** [comment]
Since the changes done on analysis-model have been released, do i also need to do any changes on analysis-model-api for the version or is it updated already? 
Is it correct that the most recent version of analysis-model-api that also includes our changes is 10.22.0 (what should be reference in pom.xml of warnings-ng-plugin)?

**Comment by uhafner** [comment]
I updated the pom.xml so you need to rebase and retry.

**Comment by evahabeeballah** [comment]
Hi, 
Regarding the 'Enforce PR labels / enforce-label (pull_request)' check that is failed, can you please let me know what that means, or what can I do to pass the check?

**Comment by uhafner** [comment]
> Hi,
> Regarding the 'Enforce PR labels / enforce-label (pull_request)' check that is failed, can you please let me know what that means, or what can I do to pass the check?

This is a check for me so that I do not merge without a label.

File: plugin/src/main/java/io/jenkins/plugins/analysis/warnings/PolyspaceParse.java
Comment (uhafner) [inline plugin/src/main/java/io/jenkins/plugins/analysis/warnings/PolyspaceParse.java:29]: Seems that I missed that detail in my analysis-model PR.

Wouldn't that be a more appropriate symbol?

```suggestion
    @Symbol("polyspace")
```

Where does the `Parse` part came from? Is the tool called Polyspace Parse? Or is this suffix solely from your parser?

Either way: symbols need to use CamelCase, IDs need to use url-notation (all lower case). I can fix that right away in the analysis model as well. 

What do you think?

File: plugin/src/main/webapp/icons/LICENSE.txt
Comment (uhafner) [inline plugin/src/main/webapp/icons/LICENSE.txt:266]: What is the actual license? Are we allowed to add it to our MIT based project?

File: plugin/src/main/java/io/jenkins/plugins/analysis/warnings/PolyspaceParse.java
Comment (evahabeeballah) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/PolyspaceParse.java:29]: The idea was to differentiate between polyspace commands and the parser command, just in case there is a command from polyspace tool defined in the same way as 'polyspace'.

File: plugin/src/main/webapp/icons/LICENSE.txt
Comment (evahabeeballah) [reply plugin/src/main/webapp/icons/LICENSE.txt:266]: After checking the licence file, we are not sure if we can use it this way. I will remove the icon file and push the changes again, will that be okay ?

Comment (uhafner) [reply plugin/src/main/webapp/icons/LICENSE.txt:266]: Yes, this would be more safe. Image licensing is always problematic...

File: plugin/src/main/java/io/jenkins/plugins/analysis/warnings/PolyspaceParse.java
Comment (uhafner) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/PolyspaceParse.java:29]: I see. Then let us use `polyspaceParser` if that is fine for you as well, since we already have several other tools that do it this way (PMD, AndroidLint).

Comment (uhafner) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/PolyspaceParse.java:29]: The appropriate ID would then be `polyspace-parser`.

Comment (evahabeeballah) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/PolyspaceParse.java:29]: Yes polyspace-parser would be fine then. I will change that now.

Comment (uhafner) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/PolyspaceParse.java:29]: In analysis-model as well?

Comment (evahabeeballah) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/PolyspaceParse.java:29]: Yes also in analysis-model. But you can change that ? Since the pull request is already approved.

Comment (uhafner) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/PolyspaceParse.java:29]: Yes, this library is under my control and only used by tools that I control. So no worries...

Comment (evahabeeballah) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/PolyspaceParse.java]: I will also do the same for the Simulink check Parser in the other Pull request

**Comment by evahabeeballah** [comment]
Hello, I am not sure what is the problem with this pull request. Can you please let me know if there is anything to be done from my side. Thanks !

File: plugin/pom.xml
Comment (uhafner) [inline plugin/pom.xml:28]: I think you simply forgot to update the model version.

**Review by uhafner — APPROVED** [review]

**Review by uhafner — APPROVED** [review]

**Pull request merged**
